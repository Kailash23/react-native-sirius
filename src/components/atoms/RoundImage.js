import React from 'react';
import { StyleSheet } from 'react-native';
import FastImage from 'react-native-fast-image';
import { SIZES, RoundImageSize } from '@styles/sizes';
import { layout } from '@utils/Layout';
import { withTheme } from 'react-native-paper';

const width = layout.width;
const placeholder = require('../../../react/images/placeholder.png');

function RoundImage(props) {
  const { url } = props;

  return (
    <FastImage
      style={[
        {
          height: RoundImageSize,
          width: RoundImageSize,
          borderRadius: RoundImageSize / 2
        }
      ]}
      fallback
      defaultSource={placeholder}
      source={{
        uri: url,
        priority: FastImage.priority.low
      }}
      resizeMode={FastImage.resizeMode.cover}
    />
  );
}

export default withTheme(RoundImage);
