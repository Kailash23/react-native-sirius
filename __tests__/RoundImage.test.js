import RoundImage from '@atoms/RoundImage';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

configure({ adapter: new Adapter() });

describe('Testing <RoundImage/>', () => {
  test('RoundImage should render properly ', () => {
    const wrapper = shallow(
      <RoundImage url={'https://picsum.photos/200/300'} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('RoundImage should render default image if the iamge url is null or empty ', () => {
    const wrapper = shallow(<RoundImage url={null} />);
    expect(wrapper).toMatchSnapshot();
  });
});
