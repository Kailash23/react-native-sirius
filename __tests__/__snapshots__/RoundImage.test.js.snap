// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <RoundImage/> RoundImage should render default image if the iamge url is null or empty  1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  url={null}
/>
`;

exports[`Testing <RoundImage/> RoundImage should render properly  1`] = `
<ThemedComponent
  _reactThemeProviderForwardedRef={null}
  url="https://picsum.photos/200/300"
/>
`;
